src/main/java/org/broadinstitute/hellbender/tools/spark/pipelines/ReadsPipelineSpark.java

```diff
@@ -105,7 +110,10 @@ protected void runTool(final JavaSparkContext ctx) {
         }
 
         final JavaRDD<GATKRead> initialReads = getReads();
-        final JavaRDD<GATKRead> markedReadsWithOD = MarkDuplicatesSpark.mark(initialReads, getHeaderForReads(), duplicatesScoringStrategy, new OpticalDuplicateFinder());
+        if (parallelism == 0) {
+            parallelism = ReadsSparkSource.getRecommendedParallelism(ctx, getReadSourceName(), initialReads.partitions().size());
+        }
+        final JavaRDD<GATKRead> markedReadsWithOD = MarkDuplicatesSpark.mark(initialReads, getHeaderForReads(), duplicatesScoringStrategy, new OpticalDuplicateFinder(), parallelism);
```

Comment:
Is there a case for having different levels of parallelism in practice? I'm not sure there is, so I've kept a single `parallelism` option for the time being.